pkg/providers/aws/credentials.go

```diff
@@ -141,13 +142,24 @@ type CredentialManager interface {
 	ReconcileBucketOwnerCredentials(ctx context.Context, name, ns, bucket string) (*Credentials, error)
 }
 
-func NewCredentialManager(client client.Client) CredentialManager {
-	ns, _ := k8sutil.GetOperatorNamespace()
-	_, err := getSTSCredentialsSecret(context.TODO(), client, ns)
-	if errors.IsNotFound(err) {
-		return NewCredentialMinterCredentialManager(client)
+func NewCredentialManager(client client.Client) (CredentialManager, error) {
+	ns, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return nil, err
+	}
+	secret, err := getSTSCredentialsSecret(context.TODO(), client, ns)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			resources.SetSTSCredentialsSecretMetric(ns, err)
```

Comment:
I think this metric set could be moved outside of the isNotFound error as we are accepting that one as not an error effectively.

Reply:
Done 👍🏻